add schedule list to detail views

jlmitch5 opened the pull request:
this moves the ScheduleList component to the components folder, and adds it to the project, jt, and wfjt views.

Commit:
update directory structure of schedule list files

Commit:
add schedule list to proj, jt and wfjt

Commit:
add test for conditional show/hide of schedules tab for project detail view

jakemcdermott commented on awx/ui_next/src/screens/Project/Project.jsx:
```diff
@@ -134,16 +134,17 @@ class Project extends Component {
+      link: `${match.url}/job_templates`,
+    });
+
+    if (project?.scm_type && project.scm_type !== '') {
```
An empty string is falsey, so I think:

```js
if (project?.scm_type) {
```

would work just as well

jakemcdermott commented on awx/ui_next/src/components/ScheduleList/ScheduleList.jsx:
```diff
@@ -33,14 +33,17 @@ function ScheduleList({ i18n }) {
       const params = parseQueryString(QS_CONFIG, location.search);
+      const response = apiModel
+        ? apiModel.readScheduleList(resource.id, params)
+        : SchedulesAPI.read(params);
```
We don't use the `resource` or `apiModel` props for anything but this one network call. I'm wondering if the parent component should wrap up the api call in a callback and pass it as a prop:

```js
function ScheduleList({ i18n, loadSchedules }) {
   ...
    useCallback(async () => {
      const params = parseQueryString(QS_CONFIG, location.search);
      const response = loadSchedules(params);
```
This would improve the testability of the component:
  - It would remove the conditional assignment, which is effectively another code path to test. 
  - Simpler contract and fewer props overall to test
  - Fewer mocks needed for most tests cases - In some cases the function you pass in could just be a stub promise.

I guess a drawback might be a little more code in the parent component but I think it's probably worth it. Thoughts?

Commit:
update ScheduleList api read to single prop

Commit:
simplify conditional checking if schedules tab should be shown on proj detail

Commit:
assorted schedule list fixes:
- remove pagesection and card from component...move to where called for root list
- remove unnecessary placeholder schedule tab on job template detail
